service/recognition.js

```diff
@@ -211,6 +211,11 @@ async function isGratitudeAffordable(gratitude) {
     gratitude.giver.id,
     gratitude.giver.tz
   );
+  if (gratitude.giver_in_receivers) {
+    gratitude.receivers = gratitude.receivers.filter(
+      (x) => x.id !== gratitude.giver.id
+    );
+  }
```

Comment:
hmmm I question the design choice of an `is` method that modifies the state of the argument. While I cannot think of a reason we would ever need to keep the recognizer in the array of receivers and this 'fixes' this issue for anything downstream it just _feels_ bad. But functionally I get it and it fixes the issue so 👍

Reply:
I feel the same way actually, having this change the state of the arg makes me really hesitant, and I would have some tests to consider, such as testing sending a fistbump to yourself. Is there any possible pattern that results in a receivers list of length 0 for example?

It might not be an issue, but I'd be somewhat more comfortable with making a copy of the list instead.